### PR TITLE
feat: maturity facet

### PR DESCRIPTION
**Pull request by jaime-iobermudez**
This pull request introduces a significant architectural refactor to the bond maturity logic by extracting maturity-related functions from the `BondFacet` into a new dedicated `MaturityFacet`, registered under a new resolver key. This change improves modularity, separation of concerns, and maintainability, while preserving backward compatibility at the interface level. The update includes new contracts and interfaces, updates to facet registries and scripts, and comprehensive test coverage for the new facet.

**Core contract and interface refactoring:**

- Extracted `fullRedeemAtMaturity` and `updateMaturityDate` logic from `Bond.sol` and `IBondManagement.sol` into a new `MaturityFacet` (composed of `IMaturity.sol`, `Maturity.sol`, and `MaturityFacet.sol`), registered under the new `_MATURITY_RESOLVER_KEY` constant. (`[[1]](diffhunk://#diff-1f56b267a961f9c29c98d930427699d45f31abde027f448d0dddefe84e4a7c76R1-R29)`, `[[2]](diffhunk://#diff-a6761447814cc412fd1b009f087d5b68cc0a5e3ad8ef1dd4e34fdd2e6b3b465dR557-R559)`, `[[3]](diffhunk://#diff-3387d7529ee32c487d194f5cc11215c04ca4f656e77489bf5a2c8d5222555068R1-R41)`, `[[4]](diffhunk://#diff-c2454de09e3856738b6f56be23519a9bd25f8267813cfd47cf6ecfb8212a44e3R1-R73)`, `[[5]](diffhunk://#diff-cd4a45904fae3ec653e1dfd904bb6a4c43f1bed0459d4df5f8fca75de9e0e4cdR1-R42)`)
- Updated `IAsset` to inherit from `IMaturity`, ensuring maturity functions remain accessible via the main asset interface. (`[[1]](diffhunk://#diff-40dbee73667a1c027b07f3aba73c9a5f76a07c2e6cb6c8b88fe4201d84435d75R13)`, `[[2]](diffhunk://#diff-40dbee73667a1c027b07f3aba73c9a5f76a07c2e6cb6c8b88fe4201d84435d75R165)`)

**Bond and facet selector changes:**

- Removed `fullRedeemAtMaturity` and `updateMaturityDate` from `Bond.sol`, `IBondManagement.sol`, and `BondUSAFacetBase.sol` static selectors; `Bond.sol` now only exposes `redeemAtMaturityByPartition`. (`[[1]](diffhunk://#diff-f33c3345718ff83f7711727e5f266ceaaffe0faa45865baf7b9b349627dd9a82L6-L69)`, `[[2]](diffhunk://#diff-f33c3345718ff83f7711727e5f266ceaaffe0faa45865baf7b9b349627dd9a82L104-L130)`, `[[3]](diffhunk://#diff-b48b13a69faf4ee4891232f4afbaa5c883fe18156cab7599c7571b1c401ddd9bL9-L14)`, `[[4]](diffhunk://#diff-b48b13a69faf4ee4891232f4afbaa5c883fe18156cab7599c7571b1c401ddd9bL23-L28)`, `[[5]](diffhunk://#diff-551e2568b6db3a6e0f0587f3839f0b89f9ef059f151063b48d90a022fdf83be4L11-R25)`)
- Added the new `MaturityFacet` to facet registries and updated scripts to register and expose the correct selectors and interface IDs. (`[[1]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949L13-R14)`, `[[2]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R106)`, `[[3]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949L1682-L1689)`, `[[4]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949L1698-L1705)`, `[[5]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949L1878-L1885)`, `[[6]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949L1894-L1901)`)

**Testing and documentation:**

- Added comprehensive tests for `MaturityFacet` covering positive, negative, and edge cases for both `fullRedeemAtMaturity` and `updateMaturityDate`. Moved relevant tests from `bond.test.ts` and `coupon.test.ts` to the new `maturity.test.ts`. (`[.changeset/maf-maturity.mdR1-R29](diffhunk://#diff-1f56b267a961f9c29c98d930427699d45f31abde027f448d0dddefe84e4a7c76R1-R29)`)

**Constants:**

- Introduced the `_MATURITY_RESOLVER_KEY` constant in `resolverKeys.sol` for the new facet registration. (`[packages/ats/contracts/contracts/constants/resolverKeys.solR557-R559](diffhunk://#diff-a6761447814cc412fd1b009f087d5b68cc0a5e3ad8ef1dd4e34fdd2e6b3b465dR557-R559)`)

**Non-breaking change:**

- All function selectors remain unchanged, ensuring backward compatibility for calls via `IAsset`. (`[.changeset/maf-maturity.mdR1-R29](diffhunk://#diff-1f56b267a961f9c29c98d930427699d45f31abde027f448d0dddefe84e4a7c76R1-R29)`)

## Type of change

- [ ] Bug fix 🐞
- [ ] New feature ✨
- [ ] Breaking change 💥
- [ ] Documentation update 📖
- [ ] Refactor 🔧

## Testing

**Node version**:

- [ ] 20
- [ ] 22
- [ ] 24

## Checklist

- [ ] Style Guidelines followed ✅
- [ ] Documentation Updated 📚
- [ ] **Linters** - No New Warnings ⚠️
- [ ] Local Tests Pass ✅
- [ ] Effective Tests Added ✔️
- [ ] No reduction of **Coverage** ✅

### CODE REVIEW

File: packages/ats/contracts/contracts/facets/maturity/Maturity.sol [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            _checkUnexpectedError(balance == 0, KPI_BOND_REDEEM_BALANCE);
+            ERC1410StorageWrapper.redeemByPartition(
+                partition,
+                _tokenHolder,
+                EvmAccessors.getMsgSender(),
+                balance,
+                "",
+                ""
+            );
+        }
```
Comment (marcosio) [inline packages/ats/contracts/contracts/facets/maturity/Maturity.sol:53]: for is not optimised. Use  unchecked  {--i;}, create  an address sender = EvmAccessors.getMsgSender() out of the for, must declare a uint256 length = partitions.length;

**Review by marcosio — CHANGES_REQUESTED** [review]

**Review by marcosio — APPROVED** [review]
